src/ipld/selector/mod.rs

```diff
@@ -209,7 +209,7 @@ impl Selector {
             ExploreFields { mut fields } => match ipld {
                 Ipld::Map(m) => {
                         m.get(p)?;
-                        fields.remove(p)
+                        fields.swap_remove(p)
```

Comment:
As per the [doc](https://docs.rs/indexmap/latest/indexmap/map/struct.IndexMap.html#method.remove)
> This is equivalent to [.swap_remove(key)](https://docs.rs/indexmap/latest/indexmap/map/struct.IndexMap.html#method.swap_remove)

Just to confirm, the order does not matter here, right? @lemmih

Reply:
seems to be looking good